hive/src/main/java/io/confluent/connect/storage/hive/HiveSchemaConverter.java

```diff
@@ -77,51 +79,58 @@ public static List<FieldSchema> convertSchemaMaybeLogical(Schema schema) {
     return columns;
   }
 
-  public static TypeInfo convert(Schema schema) {
+  public static TypeInfo convert(Schema schema, boolean convertLogical) {
     // TODO: throw an error on recursive types
     switch (schema.type()) {
       case STRUCT:
-        return convertStruct(schema);
+        return convertStruct(schema, convertLogical);
       case ARRAY:
-        return convertArray(schema);
+        return convertArray(schema, convertLogical);
       case MAP:
-        return convertMap(schema);
+        return convertMap(schema, convertLogical);
       default:
-        return convertPrimitive(schema);
+        return convertLogical ? convertPrimitiveMaybeLogical(schema) : convertPrimitive(schema);
```

Comment:
hi, thanks for the PR. 
QQ - is there a reason we can't just call `convertPrimitiveMaybeLogical(schema)` with `default` in `convert(Schema schem)` ? 
IIUC, unless schema.name() matches any of `Decimal`, `TypeInfoFactory.dateTypeInfo` OR `TypeInfoFactory.timestampTypeInfo ` -> convertPrimitiveMaybeLogical(schema) will call convertPrimitive(schema) internally - so should be the same. 
ie why do we need `convertSchemaMaybeLogical` and `convertSchema` both ?

Reply:
the above question is not really related to the change though. 

The change looks good, optionally a cleaner solution could be to create a class that implements these `convert` functions. And if initialized with logcial=true, then just overrides default behvaiour to call convertPrimitiveMaybeLogical instead of convertPrimitive. We don't really need to pass the variable to each step in the recursion  (given that convertMap, convertStruct etc doesn't really use it themselves)

Reply:
We do need both, right now. Logical types are internally represented with primitives. For example Time, Date, Timestamps are `int`, Decimal is `byte[]`. `convertPrimitive` would result in representation of these as the primitives in hive (int, byte[]) and this actually fails for Orc as while writing we write the logical type. Moving to `convertSchemaMaybeLogical` for everything is something we could do but most places use the primitive approach and it may require additional changes downstream to switch. The changes to Orc is being done in a separate PR in hdfs.

Creating another class makes for better code but will require new instantiations for every schema conversion. That induces requirements on how the API, for example if the schema generated is not being cached or changing frequently it might create a bunch of single use classes. It is better to keep things stateless.

Reply:
gotcha! 
thanks!